### ISSUE TITLE
Encoder automatically encodes in 24 fps.  Can this be changed?

### ISSUE THREAD

**Issue by dxbender** [WWBN/AVideo#7796]
I've noticed that encoded videos drop frames.  I finally decided to investigate and discovered that files are encoded in 24 fps.  Well, I produce videos in 25 and 30 fps.  Obviously 1 or 6 frames are dropped.

How can this be set to automatically adjust to the right frame rate OR manually set to the frame rate of the video being encoded?

**Comment by dxbender** [comment]
I may have figured it out...
-c:v h264 -vf scale=-2:{$resolution} -r 24 -g 48 -keyint_min 48 -sc_threshold 0 -bf 3 -b_strategy 2 -minrate {$minrate}k -crf 23 -maxrate {$maxrate}k -bufsize {$bufsize}k -c:a aac -b:a {$autioBitrate}k -f hls -hls_time 6 -hl

The switch "-r 24" is frame rate?  If so, changing this to 25 should fix my issue?  Is this the only place this needs to be adjusted?  Would removing the switch make it automatic

**Comment by ronaldod** [comment]
yes you can just look at [ffmpeg documentation](https://ffmpeg.org/ffmpeg.html) and -r 24 means forcing the output to 24 fps.

**Comment by walt93** [comment]
Pro tip: Ask Chat GPT 4 to review your parameters, then verify (sometimes it *will* throw you forcibly into a ditch), for shit like this it's just going to explain the documentation to you.

That being said, I got it to build a complete task scheduler app using  Python/Flask/MySQL/SQLAlchemy/RabbitMQ and even added some AI to it, having ChatGPT summarize the title and description to fit my Roku channel, and had ChatGPT bootstrap itself, into the app, writing the code to interface with its API.

One of these days I have to sic' it on AVideo… ;-)
